### DISCUSSION TITLE
Memory Usage [0.15.0 Suggestion]

### DISCUSSION THREAD

**Discussion by JohnSh00ter** [ssilverman/QNEthernet#10] [Ideas]
Suggestion for 0.15.0:

Hi I'm using your Library in combination with LVGL for a touchdisplay.
After moving a few things into Flash/Ram2 I have only 32k left in Ram1.

Would it be possible to move the library's variables from RAM1 to RAM2 using a marco or something.
Perhaps: #define QN_MEM DMAMEM or something like that?

Anyways, great work!
Nothing else comes even close to your library.

**Comment by ssilverman** [comment]
Yes, I agree that the memory usage could stand to be improved. It’s something I’m thinking about.

**Comment by ssilverman** [comment]
The latest v0.17.0 moves 15k of Ethernet buffers into RAM2. See if this helps your app.

**Comment by ssilverman** [comment]
The lwIP memory was moved to RAM2 by default in v0.19.0.

**Comment by ssilverman** [comment]
lwIP memory has been changed to use the system's `malloc` by default. This is equivalent to putting it in RAM2, but not contiguously. This can be changed.

See: https://github.com/ssilverman/QNEthernet/tree/a8da091f25d99f0aba06d79b2b5693966194167a#heap-memory-use
